window_manager/src/MsWindow.cpp

```diff
@@ -0,0 +1,177 @@
+		PostQuitMessage(0);
+		stopWindowLoop();
+	}
+	return false;
+
+	case WM_PAINT: {
+		PAINTSTRUCT ps;
+		HDC hdc = BeginPaint(window_handle_, &ps);
+		const auto &bg_color = config_.background_color;
+		HBRUSH brush = CreateSolidBrush(RGB(bg_color.r() * 255, bg_color.g() * 255, bg_color.b() * 255)); // TODO: add functionality to RGBAColor
+		FillRect(hdc, &ps.rcPaint, brush);
+
+		/*COLORREF color = RGB(255,255,0);
+		SetConsoleTitle(L"Pixel In Console?");
+		for( int i = 0 ; i < 50 ; i++ )
+		{
+			SetPixel(hdc, 5+i, 12, color); // SetPixel(HDC hdc, int x, int y, COLORREF color)
+		}*/
+
+		EndPaint(window_handle_, &ps);
+		DeleteObject(brush);
+	}
+	return false;
+
+	case WM_COMMAND: {
+		
+	}
+	return false;
+
+	default:
+		return DefWindowProc(window_handle_, we.type, we.wparam, we.lparam);
+	}
+	return true;
+}
+
+LRESULT MsWindow::handleMessage(UINT msg, WPARAM wparam, LPARAM lparam) {
+	switch (msg)
+	{
+	case WM_DESTROY: {
+		PostQuitMessage(0);
+		stopWindowLoop();
+	}
+	return false;
+
+	case WM_PAINT: {
+		PAINTSTRUCT ps;
+		HDC hdc = BeginPaint(window_handle_, &ps);
+		const auto &bg_color = config_.background_color;
+		HBRUSH brush = CreateSolidBrush(RGB(bg_color.r() * 255, bg_color.g() * 255, bg_color.b() * 255)); // TODO: add functionality to RGBAColor
+		FillRect(hdc, &ps.rcPaint, brush);
+		EndPaint(window_handle_, &ps);
+		DeleteObject(brush);
+	}
+	return false;
+
+	case WM_COMMAND: {
+		
+	}
+	return false;
+
+	default:
+		return DefWindowProc(window_handle_, msg, wparam, lparam);
+	}
+	return true;
+}
+
+void MsWindow::windowLoop() {
+	MSG msg { };
+	while (GetMessage(&msg, nullptr, 0, 0) && running_.load())
+	{
+		TranslateMessage(&msg);
+		DispatchMessage(&msg);
+	}
+}
+
+PCWSTR MsWindow::windowClassName() {
+	return L"Windows WinAPI Window";
+}
+
+void MsWindow::setWindowHandle(HWND hwnd) {
+	window_handle_ = hwnd;
+}
+
+HWND MsWindow::getWindowHandle() const {
+	return window_handle_;
+}
+
+HDC MsWindow::getGraphicContext() const {
+	return GetDC(window_handle_);
+}
+
+void MsWindow::kill() {
+	stopWindowLoop();
+	DestroyWindow(window_handle_);
+}
+
+void MsWindow::onConfigure(window::ConfigureHandlerType &&handler) {
+    configure_handler_ = std::move(handler);
+}
+
+void MsWindow::onUnmap(window::UunmapHandlerType &&handler) {
+    unmap_handler_ = std::move(handler);
+}
+
+rcbe::rendering::RenderingContextPtr MsWindow::getRenderingContext() const {
+	return rendering_context_;
+}
+
+const window_config &MsWindow::getConfig() const {
+    return config_;
+}
+
+const std::shared_ptr<rcbe::core::AbstractInputManager> &MsWindow::getInputManager() const {
+	return input_manager_;
+}
+
+void MsWindow::setInputManager(const std::shared_ptr<rcbe::core::AbstractInputManager> &manager) {
+	input_manager_ = manager;
+}
+
+void MsWindow::setRenderer(rdmn::render::RendererPtr renderer_ptr) {
+	renderer_ = std::move(renderer_ptr);
+}
+
+const rdmn::render::RendererPtr &MsWindow::getRenderer() const {
+	return renderer_;
+}
```

Comment:
I really believe renderer shouldn't be a part of window, rather there should be a concept or an interface, for such entity, or maybe even, it's enough to just have `RenderingContext`.

Reply:
Will refactor in separate task.